src/test/java/io/supertokens/test/GetUsersByEmailAPITest2_8.java

```diff
@@ -0,0 +1,150 @@
+/*
```

Comment:
You also need a test in ThirdParty.java to test the raw functions properly:
- If you use the get users by email function with no users, it should return an empty array
- If you provide an email that doesn't exist (but there are users), it should return an empty array
- If you have multiple users with different emails, and you use one email, it should return the relevant subset of users.

Again, please see how other functions + APIs are tested..

Reply:
> You also need a test in ThirdParty.java to test the raw functions properly:

I don't know what you mean by raw functions. Do you mean to unit test them? I'm all for it, but there's no other test I could find that did that. 

> * If you use the get users by email function with no users, it should return an empty array
> * If you provide an email that doesn't exist (but there are users), it should return an empty array
> * If you have multiple users with different emails, and you use one email, it should return the relevant subset of users.

This makes sense.

Reply:
By raw function, I meant The function that the API uses (`ThirdParty. getUsersByEmail`).. sorry, should have been more clear about this in my prev comment.

Reply:
But you don't need to do this i guess cause you already implemented these tests via the API itself.